brave/brave-core
# Cleans up pre-populated search engines.

Engines kept:
- Google
- Duckduckgo
- Qwant
- Bing
- Startpage.

Adds a unit test to verify the search engines overridden by us are used
instead of the original ones.

Lint fixes.

Fixes brave/brave-browser#3316

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:

**Automated testing:**
`npm run test -- brave_unit_tests --filter=BraveTemplateURLPrepopulateDataTest.*`
`npm run test -- brave_unit_tests --filter=BraveTemplateURLServiceUtilTest.*`

**Manual testing:**
1. Test transition from pre-existing profile; default not changed.
- Start Brave version prior to version that contains this change,
- Navigate to brave://settings/searchEngines and verify that the default search engine has not been 
  changed (e.g. Google for U.S., Qwant for DE & FR),
- Exit Brave,
- Start Brave version that contains this change,
- Navigate to brave://settings/searchEngines and verify that only the 5 engines listed above are 
  present and that the default engine has not changed.

2. Test transition from pre-existing profile; default changed.
- Start Brave version prior to version that contains this change,
- Navigate to brave://settings/searchEngines and change the default search engine to Amazon,
- Exit Brave,
- Start Brave version that contains this change,
- Navigate to brave://settings/searchEngines and verify that only the 5 engines listed above + Amazon 
  are present and that the default engine is Amazon.

3. Test new profile.
- Start Brave version that contains this change with a fresh profile,
- Navigate to brave://settings/searchEngines and verify that only the 5 engines listed above are present 
  and that Qwant is the default engine for DE & FR and Google for all others.

4. Test import from Muon; removed engine.
- Start Brave (Muon) and change the default search engine to Amazon, exit the browser,
- Start Brave version that contains this change with a fresh profile,
- In the Welcome screen, click next, then Import,
- Pick import from Brave (old) and leave all boxes checked,
- Finish import,
- Navigate to brave://settings/searchEngines and verify that Google (or Qwant for DE, FR locales)
  remains the default engine.

5. Test import from Muon; kept engine.
- Start Brave (Muon) and change the default search engine to DuckDuckGo, exit the browser,
- Start Brave version that contains this change with a fresh profile,
- In the Welcome screen, click next, then Import,
- Pick import from Brave (old) and leave all boxes checked,
- Finish import,
- Navigate to brave://settings/searchEngines and verify that DuckDuckGo is set as the 
  default engine.

## Reviewer Checklist:
